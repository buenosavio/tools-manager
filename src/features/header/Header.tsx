import {
  ButtonAdd,
  HeaderMain,
  InputSearch,
  TitlePage,
  SubTitlePage,
  SpanSearchTags,
  SearchSection
} from './Header.styles'
import ModalNewCard from '../form/ModalNewCard'
import { useState } from 'react'
import { useDispatch } from 'react-redux'
import { toolActions } from '../../store/tools/duck'

const Header = () => {
  const dispatch = useDispatch()
  const [modalOpen, setModalOpen] = useState(false)

  return (
    <HeaderMain>
      <TitlePage>VUTTR</TitlePage>
      <SubTitlePage>Very Useful Tools to Remember</SubTitlePage>
      <SearchSection>
        <div>
          <InputSearch
            type="text"
            name="search"
            placeholder="Search..."
            onChange={e => dispatch(toolActions.search(e.target.value))}
          />
          <input
            type="checkbox"
            onClick={() => dispatch(toolActions.toggleIsOnlyTag())}
          />
          <SpanSearchTags>search in tags only</SpanSearchTags>
        </div>
        <ButtonAdd onClick={() => setModalOpen(true)}>Add</ButtonAdd>
        {modalOpen && <ModalNewCard setOpenModal={setModalOpen} />}
      </SearchSection>
    </HeaderMain>
  )
}

export default Header
