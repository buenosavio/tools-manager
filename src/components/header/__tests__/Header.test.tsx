import Header from '../Header'
import renderer from 'react-test-renderer'
import configureStore from 'redux-mock-store'
import { Provider } from 'react-redux'
import { mount } from 'enzyme'
import withProvider from '../../../utils/withContext'

describe('Header: General test of component', () => {
  const initialState = { tools: [], isOnlyTag: true, loading: true, search: '' }
  const mockStore = configureStore()
  let store = mockStore(initialState)

  const Component = withProvider(Header, store)

  const mockModalState = jest.fn()
  jest.mock('react', () => {
    return {
      useState: () => mockModalState
    }
  })

  test('renders correctly by default UI', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Header />
        </Provider>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('Modal should ne open where click on +Add button', () => {
    const componentRender = mount(<Component />)
    componentRender.find('#add-btn').at(0).simulate('click')
    expect(mockModalState).toHaveLength(0)
  })
})
