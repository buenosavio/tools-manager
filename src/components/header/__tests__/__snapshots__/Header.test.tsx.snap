// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header: General test of component renders correctly by default UI 1`] = `
<header
  className="sc-bczRLJ cdzGpq"
  id="headof"
>
  <h1
    className="sc-gsnTZi iHDvAZ"
  >
    VUTTR
  </h1>
  <h3
    className="sc-dkzDqf ICVwW"
  >
    Very Useful Tools to Remember
  </h3>
  <div
    className="sc-hKMtZM idGchu"
  >
    <div>
      <input
        className="sc-eCYdqJ gSofyM"
        name="search"
        onChange={[Function]}
        placeholder="Search..."
        type="text"
      />
      <input
        onClick={[Function]}
        type="checkbox"
      />
      <span
        className="sc-jSMfEi icURsM"
      >
        search in tags only
      </span>
    </div>
    <button
      className="sc-gKXOVf cCHAos"
      id="add-btn"
      onClick={[Function]}
    >
      Add
    </button>
  </div>
</header>
`;
